src/components/organisms/AssetActions/Trade/Swap.tsx

```diff
@@ -52,25 +53,25 @@ export default function Swap({
     validateForm
   }: FormikContextType<FormTradeData> = useFormikContext()
 
-  /// Values used for calculation of price impact
+  // Values used for calculation of price impact
   const [spotPrice, setSpotPrice] = useState<string>()
   const [totalValue, setTotalValue] = useState<string>()
   const [tokenAmount, setTokenAmount] = useState<string>()
-  ///
+
   useEffect(() => {
-    if (!ddo || !balance || !values || !price) return
+    if (!ddo || !balance || !values?.type || !price) return
 
     async function calculateMaximum() {
       const dtAmount = values.type === 'buy' ? maxDt : balance.datatoken
```

Comment:
after the number → decimal PR merge in here, this seems to be the only place left in this PR where we do calculations and we might want to use `Decimal` here but not sure if really needed. So `dtAmount` & `oceanAmount`

Reply:
cool, I'll take a look